modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/AbstractChallengeRestTest.java

```diff
@@ -0,0 +1,317 @@
+            getBaselineTemplateName(),
+            getBaselineDataStreamName() + "*",
+            baselineSettings,
+            baselineMappings,
+            getBaselineTemplatePriority()
+        );
+        assert createBaselineTemplateResponse.getStatusLine().getStatusCode() == RestStatus.OK.getStatus();
+
+        final Response createContenderTemplateResponse = createTemplates(
+            getContenderTemplateName(),
+            getContenderDataStreamName() + "*",
+            contenderSettings,
+            contenderMappings,
+            getContenderTemplatePriority()
+        );
+        assert createContenderTemplateResponse.getStatusLine().getStatusCode() == RestStatus.OK.getStatus();
+    }
+
+    private void createDataStreams() throws IOException {
+        final Response craeteBaselineDataStreamResponse = client.performRequest(
+            new Request("PUT", "_data_stream/" + getBaselineDataStreamName())
+        );
+        assert craeteBaselineDataStreamResponse.getStatusLine().getStatusCode() == RestStatus.OK.getStatus();
+
+        final Response createContenderDataStreamResponse = client.performRequest(
+            new Request("PUT", "_data_stream/" + getContenderDataStreamName())
+        );
+        assert createContenderDataStreamResponse.getStatusLine().getStatusCode() == RestStatus.OK.getStatus();
+    }
+
+    private Response createTemplates(
+        final String templateName,
+        final String pattern,
+        final Settings.Builder settings,
+        final XContentBuilder mappings,
+        int priority
+    ) throws IOException {
+        final String template = """
+            {
+              "index_patterns": [ "%s" ],
+              "template": {
+                "settings":%s,
+                "mappings": %s
+              },
+              "data_stream": {},
+              "priority": %d
+            }
+            """;
+        final Request request = new Request("PUT", "/_index_template/" + templateName);
+        final String jsonSettings = settings.build().toString();
+        final String jsonMappings = Strings.toString(mappings);
+        request.setJsonEntity(Strings.format(template, pattern, jsonSettings, jsonMappings, priority));
+        return client.performRequest(request);
+    }
+
+    private void deleteTemplates() throws IOException {
+        final Response deleteBaselineDataStream = client.performRequest(
+            new Request("DELETE", "/_data_stream/" + getBaselineDataStreamName())
+        );
+        assert deleteBaselineDataStream.getStatusLine().getStatusCode() == RestStatus.OK.getStatus();
+
+        final Response deleteContenderDataStream = client.performRequest(
+            new Request("DELETE", "/_data_stream/" + getContenderDataStreamName())
+        );
+        assert deleteContenderDataStream.getStatusLine().getStatusCode() == RestStatus.OK.getStatus();
+    }
+
+    private void deleteDataStreams() throws IOException {
```

Comment:
Names of this and above functions are switched around - this one deletes templates and the one above deletes data streams.